CMakeLists.txt

```diff
@@ -523,9 +523,15 @@ endif()
 if(NOT IOS AND CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CMAKE_OSX_ARCHITECTURES MATCHES "^(x86_64|arm64)$")
   set(CROSS_COMPILING_MACOSX TRUE)
   # We need to compile a universal protoc to not fail protobuf build
-  execute_process(COMMAND ./scripts/build_host_protoc.sh --other-flags "-DCMAKE_OSX_ARCHITECTURES=x86_64;arm64"
+  # We set CMAKE_TRY_COMPILE_TARGET_TYPE to be a static library (vs executable) in order to succeed the cmake compiler check for cross-compiling
```

Comment:
For more context, it'd be really nice to just have: 
```
execute_process(COMMAND ./scripts/build_host_protoc.sh --other-flags -DCMAKE_OSX_ARCHITECTURES="x86_64;arm64" -DCMAKE_TRY_COMPILE_TARGET_TYPE=STATIC_LIBRARY
...
```

But this causes a CMake error because it cannot parse the double quotes correctly. Out of the many alternatives I've tried, writing to a temp file was the only one that worked, but I am glad to switch to anything cleaner that also works.